### PR TITLE
Create container_resolvers_conf.xml.sample

### PR DESCRIPTION
**Pull request by pcm32**
This PR adds an example for container_resolvers_conf.xml.sample, with what it is hopefully correct documentation.

### CODE REVIEW

**Comment by pcm32** [comment]
We probably want to add some documentation on how to use this somewhere besides the main XML file, maybe in the mulled containers part.

File: config/container_resolvers_conf.xml.sample [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  <!-- <cached_mulled /> -->
+  <!-- cached_mulled: resolves container URI through bioconda -> mulled
+       automatic mapping, preferring cached containers in the accessible docker
+       engine. Requires docker engine. -->
+
+  <!-- <cached_mulled_singularity /> -->
+  <!-- cached_mulled_singularity: resolves container URI through
+       bioconda -> mulled automatic mapping, preferring cached singularity
+       containers accessible to local singularity binaries. Only works with
```
Comment (bgruening) [inline config/container_resolvers_conf.xml.sample]: singularity images?

File: config/container_resolvers_conf.xml.sample [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+       enable_beta_mulled_containers set to true in the galaxy.yaml config file.
+       -->
+
+  <!-- <mulled /> -->
+  <!-- mulled: resolves container URI through bioconda -> mulled automatic
+       mapping. Doesn't require docker or singularity, and should be the method
```
Comment (bgruening) [inline config/container_resolvers_conf.xml.sample]: `Doesn't require docker or singularity` this is a little bit confusing, imho.

File: config/container_resolvers_conf.xml.sample [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+       of choice if mulled resolution is required and the container execution is
+       handled by a container orchestration instead of docker directly. It will
+       also work with direct docker engine tool executions. -->
+
+  <!-- <build_mulled /> -->
+  <!-- build_mulled: builds a docker container locally for one or more bioconda
+       packages listed on the tools definition. -->
```
Comment (bgruening) [inline config/container_resolvers_conf.xml.sample]: `in` instead `on`

File: config/container_resolvers_conf.xml.sample [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+       
+  <!-- <build_mulled_singularity /> -->
+  <!-- build_mulled_singularity: builds a singularity container locally for one
+       or more bioconda packages listed on the tools definition. -->
```
Comment (bgruening) [inline config/container_resolvers_conf.xml.sample]: `in` or `as part of`.
So maybe `builds a local singularity container for one or more package requirements listed as part of the tools definition.`

**Review by bgruening — COMMENTED** [review]
Thanks @pcm32!

File: config/container_resolvers_conf.xml.sample (hunk 1 above)
Comment (pcm32) [reply config/container_resolvers_conf.xml.sample]: as I commented, I'm fairly ignorant in this part, so I'm not sure what would be the correct wording. Please paste here what you would want to see in those pieces of documentation and I can put it on the PR branch.

File: config/container_resolvers_conf.xml.sample (hunk 2 above)
Comment (pcm32) [reply config/container_resolvers_conf.xml.sample]: Will rephrase.

**Comment by pcm32** [comment]
Thanks for rev @bgruening!

**Comment by natefoo** [comment]
Thanks for documenting this! 

I wonder if it's worth putting something in here about the relationship with dependency resolvers? Right now they're basically two separate systems, so this could cause some confusion.

**Review by mvdbeek — APPROVED** [review]

**Review by natefoo — APPROVED** [review]

File: config/container_resolvers_conf.xml.sample [hunk 5]
```diff
@@ -8,13 +8,13 @@
 
   <!-- <cached_mulled /> -->
   <!-- cached_mulled: resolves container URI through bioconda to mulled
-       automatic mapping, preferring cached containers in the accessible docker
+       automatic mapping, preferring cached images in the accessible docker
```
Comment (mvdbeek) [inline config/container_resolvers_conf.xml.sample:11]: yeah, was hesitating about this. Galaxy doesn't do any `container` things at all wrt resolvers, I think we pretty consistently use `container` in the code where sometimes that's the wrong word :(.

Comment (natefoo) [reply config/container_resolvers_conf.xml.sample:11]: Ok, feel free to strip that commit.

Comment (mvdbeek) [reply config/container_resolvers_conf.xml.sample:11]: nah, at least here it's correct.

**Comment by mvdbeek** [comment]
Thanks @pcm32!

**Pull request merged**